frontend/src/metabase-lib/v1/metadata/Schema.ts

```diff
@@ -27,7 +27,7 @@ class Schema {
   }
 
   displayName() {
-    return titleize(humanize(this.name));
+    return this.name ? titleize(humanize(this.name)) : null;
```

Comment:
I did this change in previous DataPicker-related PRs because typing says `this.name` cannot be `null`.
Well, it can, and it breaks `29951-model-editor-results-metadata.cy.spec.js`.
I won't fix the typing in this PR since it requires some unrelated work. I'm just reverting it to what we have in `master`.